client/src/components/Ticketing/ticketingmanager/SubscriptionRedemption/SubscriptionRedemptionUtils.tsx

```diff
@@ -0,0 +1,183 @@
+  eventinstances: EventInstance[];
+}
+
+export interface EventInstance {
+  eventinstanceid: number;
+  eventid_fk: number;
+  eventdate: number;
+  eventtime: Date;
+  salesstatus: boolean;
+  totalseats: number;
+  availableseats: number;
+  detail: string | null;
+  purchaseuri: string | null;
+  ispreview: boolean;
+  ticketrestrictions: TicketRestriction[];
+}
+
+export interface TicketRestriction {
+  price: number;
+  concessionprice: number;
+  ticketlimit: number;
+  ticketrestrictionsid: number;
+  ticekettypeid_fk: number;
+  eventinstanceid_fk: number;
+  seasonticketpricedefaultid_fk: number | null;
+  description: string;
+  availabletickets: number;
+}
+
+export interface Subscription {
+  orderid_fk: number;
+  id: number;
+  subscriptiontypeid_fk: number;
+  seasonid_fk: number;
+  price: number;
+  previewonly: boolean;
+  firstname: string;
+  lastname: string;
+  email: string;
+  name: string;
+  ticketlimit: number;
+  seasonName: string;
+  ticketsredeemed: number;
+}
+
+export interface SubscriptionWithTicketItems {
+  orderid_fk: number;
+  id: number;
+  subscriptiontypeid_fk: number;
+  seasonid_fk: number;
+  price: number;
+  previewonly: boolean;
+  ticketlimit: number;
+  subscriptionticketitems: SubscriptionTicketItem[];
+}
+
+export interface SubscriptionTicketItem {
+  id: number;
+  ticketrestrictionid: number;
+  eventinstanceid: number;
+  eventid: number;
+  donated: boolean;
+  redeemed: Date | null;
+}
+
+export const useFetchSubscription = (subscriptionId: number, token: string) => {
+  const [events, setEvents] = useState<Event[]>([]);
+  const [subscription, setSubscription] =
+    useState<SubscriptionWithTicketItems>(undefined);
+  useEffect(() => {
+    const controller = new AbortController();
+    if (subscriptionId && token) {
+      getData(
+        `${process.env.REACT_APP_API_2_URL}/subscription-types/subscriptions/redemption/${subscriptionId}`,
+        ({subscription, events}) => {
+          setSubscription(subscription);
+          setEvents(events);
+        },
+        controller.signal,
+        token,
+      ).catch(() => console.error('Error Fetching Subscription'));
+    }
+    return () => controller.abort();
+  }, [subscriptionId, token]);
+  return {events, setEvents, setSubscription, subscription};
+};
+
+export interface Season {
+  seasonid: number;
+  name: string;
+  startdate: number;
+  enddate: number;
+  imageurl?: string;
+  seasonname: string;
+}
+
+export const useFetchSeasons = (query = '') => {
+  const [seasons, setSeasons] = useState<Season[]>([]);
+
+  useEffect(() => {
+    const controller = new AbortController();
+    getData(
+      `${process.env.REACT_APP_API_2_URL}/season/${query}`,
+      setSeasons,
+      controller.signal,
+    ).catch(() => console.error('Error fetching seasons'));
+    return () => controller.abort();
+  }, []);
+
+  return {seasons};
+};
+
+export const getSubscriptionSortFunction = (id: number) => {
+  switch (id) {
```

Comment:
Do you think you could have it switch based on a string like "name", "email", "season", etc for readability.

Reply:
Updated